x-pack/legacy/plugins/infra/server/routes/log_analysis/index_patterns/validate.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import Boom from 'boom';
+
+import { pipe } from 'fp-ts/lib/pipeable';
+import { fold } from 'fp-ts/lib/Either';
+import { identity } from 'fp-ts/lib/function';
+
+import { i18n } from '@kbn/i18n';
+
+import { InfraBackendLibs } from '../../../lib/infra_types';
+import {
+  LOG_ANALYSIS_INDEX_PATTERNS_VALIDATE_PATH,
+  indexPatternsValidateRequestPayloadRT,
+  indexPatternsValidateResponsePayloadRT,
+  IndexPatternValidateError,
+} from '../../../../common/http_api';
+
+import { throwErrors } from '../../../../common/runtime_types';
+
+export const initIndexPatternsValidateRoute = ({ framework }: InfraBackendLibs) => {
+  framework.registerRoute({
+    method: 'POST',
+    path: LOG_ANALYSIS_INDEX_PATTERNS_VALIDATE_PATH,
+    handler: async (req, res) => {
+      const payload = pipe(
+        indexPatternsValidateRequestPayloadRT.decode(req.payload),
+        fold(throwErrors(Boom.badRequest), identity)
+      );
+
+      const { timestamp, indexPatternName } = payload.data;
+      const indexPatterns = indexPatternName.split(',');
+      const errors: IndexPatternValidateError[] = [];
+
+      // Query each pattern individually, to map correctly the errors
+      await Promise.all(
+        indexPatterns.map(async indexPattern => {
+          const indices = await framework.callWithRequest(req, 'indices.get', {
```

Comment:
As it is written this doesn't support field aliases. Their indirection would have to be taken into account.

How about we use the `_field_caps` api instead, which handles field aliases for us?

Reply:
Did you mean index aliases? I'm not sure I understand how field aliases affect here (then again, I'm still new with ES, so please bear with me)

Reply:
There is an [`alias` field type](https://www.elastic.co/guide/en/elasticsearch/reference/7.5/alias.html), which can point to a different field in the mapping. That means that the `@timestamp` field might be an alias pointing to a different field, which in turn has the `date` type. The [`_field_caps` API](https://www.elastic.co/guide/en/elasticsearch/reference/7.5/search-field-caps.html) already resolves the aliases.